tsconfig.json

```diff
@@ -0,0 +1,33 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    "target": "ES5",
+    "module": "CommonJS",
+    "lib": ["ES2018"],
+    // "allowJs": true,
+    // "checkJs": true,
+    "declaration": true,
+    "outDir": "./build/",
+    "noEmit": true,
+    "isolatedModules": true,
+
+    /* Strict Type-Checking Options */
+    "strict": true,
+
+    /* Additional Checks */
+    "noUnusedLocals": true,
+    "noImplicitReturns": true,
+    "noFallthroughCasesInSwitch": true,
+
+    /* Module Resolution Options */
+    "moduleResolution": "node",
+    // "allowSyntheticDefaultImports": true,
+    "esModuleInterop": true,
```

Comment:
can we turn this off? we might need to copy over https://github.com/facebook/jest/blob/master/scripts/babel-plugin-jest-replace-ts-require-assignment.js

Reply:
Yes. Actually this flag does not affect anything since we are using babel to perform compilation anyways (`esModuleInterop` controls whether to emit a helper function for default-imports & asterik-imports when using `tsc` to compile). I enabled it because it was default when I ran `tsc --init`. I'll remove it.

Reply:
Almost, it's needed if we import modules that use `export =` in their types as well, meaning we must use `import t = require('t');` which is not supported by babel. Which is why I linked the custom babel plugin 🙂 If it work without it I'm happy, tho!